src/components/dataTable/DataTable.js

```diff
@@ -61,6 +63,13 @@ const DataTable = ({
     return filters;
   };
 
+  let disabled = false;
+
+  const setDisabled = (value) => {
+    disabled = value;
+    return true;
+  };
```

Comment:
Tried using useState, but leading to infinite renders

Reply:
I _think_ you could just initialize the variable like this:

`  let disabled = paginationObj.options?.totalSize > ROW_THRESHOLD;`

It would reinitialize each render and then you would not need the setter function or the logic inline with the JSX.